Sources/MongoSwift/BSON/BSONValue.swift

```diff
@@ -558,6 +561,9 @@ public struct ObjectId: BSONValue, Equatable, CustomStringConvertible, Codable {
     /// - SeeAlso: https://github.com/mongodb/specifications/blob/master/source/objectid.rst
     public init(fromString oid: String) {
         self.oid = oid
+        var oid_t = bson_oid_t()
+        bson_oid_init_from_string(&oid_t, oid)
+        self.timestamp = TimeInterval(bson_oid_get_time_t(&oid_t))
```

Comment:
[TimeInterval](https://developer.apple.com/documentation/foundation/timeinterval) doesn't seem like the correct type. It appears to be an alias of Double and the docs report that it "does not specify a unique point in time, or even a span between specific times." The BSON timestamp is very clearly seconds since the Unix epoch.

Is there another time-related type we could use for this? Otherwise, perhaps an integer (optionally 32-bit) makes sense.

Reply:
Why not? I think there's a good arg to be made for a UInt32, but, Swift date objects produce timestamps as `TimeInterval`s:

```swift
let time: TimeInterval = Date().timeIntervalSince1970
```

I'm amenable to either one. The Java drivers uses an `int`. I think `TimeInterval` would be more idiomatic to Swift, but a UInt32 would be simpler.

Reply:
I didn't realize TimeInterval was idiomatic for reporting offsets from the Unix epoch.

Neither the BSON spec nor the [ObjectID docs](https://docs.mongodb.com/manual/reference/method/ObjectId/) clarify whether the time component is signed. [`bson_oid_get_time_t()`](http://mongoc.org/libbson/current/bson_oid_get_time_t.html) returns a `time_t`, which is likely typedef-ed as a signed 32-bit int on most platforms. Ultimately, the drivers [ObjectID spec](https://github.com/mongodb/specifications/blob/master/source/objectid.rst#timestamp-field) does say this is an unsigned 32-bit integer, so I think that takes priority. If TimeInterval is signed, then perhaps Uint32 makes the most sense.